subsys/bluetooth/lib/CMakeLists.txt

```diff
@@ -0,0 +1,3 @@
+# SPDX-License-Identifier: Apache-2.0
+
+add_subdirectory_ifdef(CONFIG_BT_ENCRYPTED_AD_DATA encrypted_ad_data)
```

Comment:
Until the complexity of this new `subsys/bluetooth/lib/` directory increases, I'd just place the c-file directly there, rather than a subdirectory (and probably call it `ead.c` to be consistent with the API naming). We can always refactor this later into subdirectories if/when the complexity grows too large.

Reply:
I don't agree. The point of my tirades in this PR is that the host is already complex enough, exactly because of this tendency that we have in Zephyr to force complexity downwards instead of having libraries/modules on top of simple subsystems. Seemingly simple features added over time with blurry boundaries have put us in this situation.

This PR strays from the way its done currently for good reason. When we have this lib/ folder, it will be easier to migrate stuff from the host into it.

Reply:
@jori-nordic did you perhaps misunderstand what I wrote? I'm 100% in favour of putting this in a new `subsys/bluetooth/lib/` directory, rather than `subsys/bluetooth/host/`. My point was that as long as the only content of `subsys/bluetooth/lib/ead/` is a single c-file, and the only content of `subset/bluetooth/lib` is a single `ead` subdirectory then I'd just put the c-file in `subsys/bluetooth/lib/ead.c` for now. Once we have more than, say 10 c-files, under lib, or if one library consists of multiple c-files, then we can group them under their own subdirectories as `subsys/bluetooth/lib/foo/`

Reply:
Yep I did, sorry. I agree with what you said.